src/components/Requirements.vue

```diff
@@ -248,10 +252,60 @@ export default Vue.extend({
     activateMinor(id: number) {
       this.displayedMinorIndex = id;
     },
+    createCourse(course: FirestoreSemesterCourse, isRequirementsCourse: boolean) {
+      this.$emit('createCourse', course, isRequirementsCourse);
+    },
     getRequirementsTooltipText() {
       return `<b>This is your Requirements Bar <img src="${clipboard}"class = "newSemester-emoji-text"></b><br>
           <div class = "introjs-bodytext">To ease your journey, we’ve collected a list of course
           requirements based on your college and major :)</div>`;
+    },
+    getCurrentType() {
+      let currentType;
+      const currentMonth = new Date().getMonth();
+      if (currentMonth === 0) {
+        currentType = 'WI';
+      } else if (currentMonth <= 4) {
+        currentType = 'SP';
+      } else if (currentMonth <= 7) {
+        currentType = 'SU';
+      } else {
+        currentType = 'FA';
+      }
+      return currentType;
+    },
+    getCurrentYearSuffix() {
+      // If current year is 2020, get string '20'
+      const currentYear = new Date().getFullYear();
+      return currentYear.toString().substring(2);
+    },
+    getRostersFromLastTwoYears() {
+      // If current roster is FA20, get all rosters thru FA18
+      const currentType = this.getCurrentType();
+      const currentYearSuffix = this.getCurrentYearSuffix();
+
+      const currentRoster = currentType + currentYearSuffix;
+      const numRosters = 9; // Number of most recent rosters we want
+      const mostRecentRosters = [currentRoster];
+      let roster = currentRoster;
+      while (mostRecentRosters.length < numRosters) {
+        // Go backwards until we hit 2 years and add to list of rosters
+        let rosterType = roster.substring(0, 2);
+        let rosterYear = roster.substring(2);
+        if (rosterType === 'FA') {
+          rosterType = 'SU';
+        } else if (rosterType === 'SU') {
+          rosterType = 'SP';
+        } else if (rosterType === 'SP') {
+          rosterType = 'WI';
+        } else if (rosterType === 'WI') {
+          rosterType = 'FA';
+          rosterYear = (parseInt(rosterYear, 10) - 1).toString();
+        }
+        roster = rosterType + rosterYear;
+        mostRecentRosters.push(roster);
+      }
+      return mostRecentRosters;
```

Comment:
also looks like a candidate to move into `utilities.ts` to avoid having too many non-UI logic inside Requirement.vue

Reply:
Would also be helpful for @benjamin-shen 's code